### ISSUE TITLE
Type check failure for interface 

### ISSUE THREAD

**Issue by ilyub** [microsoft/TypeScript#44723]
# Bug Report

### 🔎 Search Terms

type check failure for interface

### 🕗 Version & Regression Information

Typescript 4.3.4

### ⏯ Playground Link

https://www.typescriptlang.org/play?ts=4.3.4#code/LAKA3qAEkJYHYBcCmAnAZgQwMZMgQUghGmjQAoAPALkgGcEV4BzAShoDcB7GAEwG4okAL6hBWADYZatSACFCg6AAcArgCNxMLJHLVIAInYZxKpPraQuvQiOLDRdtCrhYEMTnB2UaeC1Z42DqRkcEgA7nJkLCwCILagRNAIAJ5KuAQAvJDedAzMLJAZAHyW3PxBOs6u7p5qOYbGpuYcZYEggk4ubh5eer4t1mDxjmRqMaC2QA

### 💻 Code

```ts
{
  interface A {
    f(x: string): void;
  }

  class B {
    public f(x: "value"): void {}
  }

  function f(x: A): void {}

  f(new B());
}

{
  type A = (x: string) => void;

  function b(x: "value"): void {}

  function f(x: A): void {}

  f(b);
}
```

### 🙁 Actual behavior

Typescript accepts this code:
f(new B());
though it shows error for:
f(b);

### 🙂 Expected behavior

Typescript should show error for this code:
f(new B());
for the same reason why it shows it for:
f(b);

In both cases, function f may pass ANY string as x argument, while x argument accepts only "value" string.

**Comment by whzx5byb** [comment]
Methods (`foo(x: T): any`) are bivariant, but Function Properties (`foo: (x: T) => any`) are not, under `--strictFunctionTypes`.
See 
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-6.html 
and 
https://github.com/Microsoft/TypeScript/wiki/FAQ#why-are-function-parameters-bivariant

To produce your expected behavior, just change the interface `A` to:
```ts
interface A {
    f: (x: string) => void;
}
```

**Comment by fatcerberus** [comment]
Methods are bivariant, even with `strictFunctionTypes`.  This is working as intended and documented:
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-6.html#strict-function-types

See, for example #44592

**Comment by ilyub** [comment]
Thanks for explanation. I followed your advice to use arrow syntax.
